@here/harp-map-controls/lib/CameraKeyTrackAnimation.ts

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { GeoCoordinates, GeoCoordLike } from "@here/harp-geoutils";
+import { LookAtParams, MapView, MapViewEventNames, MapViewUtils } from "@here/harp-mapview";
+import { LoggerManager } from "@here/harp-utils";
+import THREE = require("three");
+
+const logger = LoggerManager.instance.create("CameraKeyTrackAnimation");
+
+export interface ControlPointOptions extends Partial<LookAtParams> {
+    /**
+     * @override
+     */
+    target: GeoCoordinates | GeoCoordLike;
+    /**
+     * A timestamp for the [[ControlPoint]], describes when during the animation the
+     * [[ControlPoint]] should be reached, in seconds
+     */
+    timestamp: number;
+    /**
+     * @override
+     */
+    tilt?: number;
+    /**
+     * @override
+     */
+    heading?: number;
+    /**
+     * @override
+     */
+    distance?: number;
+    /**
+     * A name for the ControlPoint
+     */
+    name?: string;
+}
+/**
+ * A ControlPoint used to defined the Track of the [[CameraKeyTrackAnimation]]
+ */
+export class ControlPoint
+    implements Pick<LookAtParams, "target" | "tilt" | "heading" | "distance"> {
+    /**
+     * The time, when this CameraPose should be reached in the animation flow, in seconds.
+     */
+    timestamp: number = 0;
+    target: GeoCoordinates;
+    name?: string;
+    tilt: number;
+    heading: number;
+    distance: number;
+
+    constructor(options: ControlPointOptions) {
```

Comment:
So as far as I can see, the difference between ControlPoint and ControlPointOptions is, that everything is defined, but not sure if this is even correct.
I would assume if I don't define the tilt for a control point the previous tilt is used in this control point. Below it is enforced to be 0

Reply:
In the current implementation the ControlPoints don't know of each other, the Animation expects a "complete" ControlPoint for each step.

Reply:
It's not so much about the control points knowing each other but to allow control points to be incomplete.
If the user for example just want's to animate the target and all other parameters should stay untouched it's a bit clumsy to use.
You have to get the current parameters like tilt and heading first and then assign it to every control points.
You also have to do this every time instead of just having one constant curve defined in the beginning.

Reply:
For the current use cases, the separate controlpoints worked fine, as they were basically recorded from an ongoing scene and then combined, but I get the point, it might be nice to modify the 3 keytracks for target, rotation and distance in a more independent manner. I propose to do this in a follow up and maybe add some timeline example using it.